eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/Utils.java

```diff
@@ -183,6 +189,46 @@ static BomDependencyNoVersion toBomDependencyNoVersion(BomDependency bomDependen
     }
 
     static List<BomDependency> parsePomFileContent(Reader responseStream) {
+        List<BomDependency> bomDependencies = new ArrayList<>();
+
+        ObjectMapper mapper = new XmlMapper();
+        try {
+            HashMap<String, Object> value = mapper.readValue(responseStream, HashMap.class);
```

Comment:
Instead of using `Map`s and `List`s everywhere would it be more intuitive to read the POM into an XML tree and perform tree-walking given that POMs should have standard formats.